src/topology/alexandroff.lean

```diff
@@ -0,0 +1,319 @@
+/-
+Copyright (c) 2021 Yourong Zang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yourong Zang
+-/
+import topology.separation
+import topology.opens
+
+/-!
+# The Alexandroff Compactification
+We construct the Alexandroff compactification of an arbitrary topological space `X` and prove
+some properties inherited from `X`.
+
+## Main defintion
+* `alexandroff`: the Alexandroff compactification
+* `of`: the inclusion map defined by `option.some`. This map requires the argument
+        `topological_space X`
+* `infty`: the extra point
+
+## Main results
+* The topological structure of `alexandroff X`
+* The connectedness of `alexandroff X` for a noncompact, preconnected `X`
+* `alexandroff X` is `T₁` for a T₁ space `X`
+* `alexandroff X` is Hausdorff if `X` is locally compact and Hausdorff
+-/
+
+noncomputable theory
+open filter set
+open_locale classical topological_space filter
+
+section basic
+
+/-- The Alexandroff extension of an arbitrary topological space `X` -/
+@[nolint unused_arguments]
+def alexandroff (X : Type*) [topological_space X] := option X
+
+variables {X : Type*} [topological_space X]
+
+/-- The embedding of `X` to its Alexandroff extension -/
+def of : X → alexandroff X := some
+
+/-- The range of the embedding -/
+def range_of (X : Type*) [topological_space X] : set (alexandroff X) := of '' (univ : set X)
```

Comment:
Is there an advantage to having this vs just using `set.range of`?

Reply:
Nope. Changed the definition to `range (@of X _)`. Is it possible to golf the new proof of `connected_space_alexandroff`? I have to use a `rw` in that proof because there is no lemma `is_preconnected_range` in mathlib.